### ISSUE TITLE
clippy does not warn me of shadowing

### ISSUE THREAD

**Issue by LucianoBestia** [rust-lang/rust-clippy#3979]
clippy 0.0.212 (1fac3808 2019-02-20)
I use:
    clippy::shadow_reuse,
    clippy::shadow_same,
    clippy::shadow_unrelated,

I have this code:
    let users = Arc::new(Mutex::new(HashMap::new()));
    // Turn our "state" into a new Filter...
    let users = warp::any().map(move || users.clone());

Clippy does not warn me about this shadowing.
https://github.com/LucianoBestia/mem2_server

**Comment by skywhale** [comment]
Another false negative example. A bug introduced during a merge conflict resolution. It would be ideal that clippy had caught it.

```rust
let mut consumers = Vec::new();
consumers.push(Consumer::new("this"));

...

// This was inserted by a wrong merge conflict resolution
let mut consumers = Vec::new();

...

consumers.push(Consumer::new("that"));
let _ = Watcher::new(consumers);  // expecting to see both "this" and "that"
```

**Comment by skywhale** [comment]
@LucianoBestia Let me suggest updating the title to "False negatives from shadow_unrelated rule" for extra clarity. Thanks for making the issue.

**Comment by flip1995** [comment]
> Known Problems: This lint, as the other shadowing related lints, currently only catches very simple patterns.

It looks like that these lints need some work.
